src/charm.py

```diff
@@ -352,6 +353,37 @@ def _on_relation_changed(self, _: ops.HookEvent, charm_state: CharmState) -> Non
         if self.get_main_unit() != self.unit.name:
             self.change_config(charm_state)
 
+    def _generate_nginx_config(self) -> None:
+        """Generate NGINX configuration based on templates."""
+        container = self.unit.get_container(synapse.SYNAPSE_NGINX_CONTAINER_NAME)
+        container.exec(
+            [
+                "cp",
+                "/etc/nginx/main_location.conf.template",
+                "/etc/nginx/main_location.conf",
+            ],
+        ).wait()
```

Comment:
That's a new one, I think previously we used some internal _exec function (that wraps container.exec) and I'm seeing a wait() for the first time.

Reply:
Well, this is an experiment ;-)
I thought about not catching the error if something happens here.

Reply:
Based:
https://discourse.charmhub.io/t/handling-temporary-errors-in-charms/13247